src/lib/cfa/ntp_conf.rb

```diff
@@ -0,0 +1,387 @@
+  # peer 192.168.1.30
+  #
+  # Order of lines is not important, except for the
+  # fudge option that should immediately follow
+  # appropriate "server" option.
+  #
+  # Important keys (that are to be read and written) are
+  # server, peer, fudge, broadcast and broadcastclient.
+  class NtpConf < ::CFA::BaseModel
+    PARSER = CFA::AugeasParser.new("ntp.lns")
+
+    PATH = "/etc/ntp.conf".freeze
+
+    RECORD_ENTRIES = %w(
+      server
+      peer
+      broadcast
+      broadcastclient
+      manycast
+      manycastclient
+      fudge
+      restrict
+    ).freeze
+
+    COLLECTION_KEYS = (RECORD_ENTRIES + %w(#comment action)).freeze
+
+    def initialize(file_handler: nil)
+      super(PARSER, PATH, file_handler: file_handler)
+    end
+
+    # Loads and fixes AugeasElement keys.
+    #
+    # AugeasParser returns collection entries with
+    # key ending in '[]', for example 'server[]'.
+    #
+    # If only exists one entry of the collection, parser
+    # return key without '[]'.
+    #
+    # For example:
+    #
+    # server 1.pool.ntp.org iburst
+    # server 2.pool.ntp.org
+    # peer 128.100.0.45
+    #
+    # In this case, key for peer entry is 'peer'.
+    # It is necessary to add '[]' to avoid possible
+    # write issues when more entries of that type
+    # are added.
+    def load
+      super
+      fix_keys
+    end
+
+    # Obtains a collection that represents the
+    # entries of the file.
+    # @return [CollectionRecord] collection to
+    #   manipulate ntp entries.
+    #
+    # The collection preserves the order of the
+    # ntp entries in the file and only contains the
+    # entries of interest (see RECORD_ENTRIES).
+    def records
+      RecordCollection.new(data)
+    end
+
+    # Obtains raw content of the file.
+    # @return [String]
+    def raw
+      PARSER.serialize(data)
+    end
+
+  private
+
+    def fix_keys
+      fix_collection_keys(data.data)
+      data.data.each do |entry|
+        if entry[:value].is_a?(AugeasTreeValue)
+          fix_collection_keys(entry[:value].tree.data)
+        end
+      end
+    end
+
+    def fix_collection_keys(entries)
+      entries.each do |entry|
+        if COLLECTION_KEYS.include?(entry[:key])
+          entry[:key] += "[]" unless entry[:key] =~ /.*\[\]$/
+        end
+      end
+    end
+
+    # class to manage ntp entries as a collection.
+    #
+    # Each element of the collection is a subclass of Record.
+    # A Record object is a wrapper of an AugeasElement.
+    class RecordCollection
+      include Enumerable
+
+      def initialize(augeas_tree)
+        @augeas_tree = augeas_tree
+      end
+
+      def each
+        record_entries.each do |augeas_element|
+          yield Record.new_from_augeas(augeas_element)
+        end
+      end
+
+      # Adds a new Record object to the collection.
+      # @param [Record] record
+      def add(record)
+        @augeas_tree.data << record.augeas
+      end
+
+      alias_method :<<, :add
```

Comment:
what is reason to have two same methods? I do not see in Array ( as this class acts like array ) method add. So I prefer to have only `<<` and if you really need additional method, then better one is `push` but it allows multiple arguments.